Add instructions for using a custom VCL snippet to bypass Fastly (#7159)
magento/devdocs: src/cloud/cdn/fastly-vcl-bypass-to-origin.md

---
group: cloud-guide
title: Bypass Fastly
redirect_from:
   - /cloud/configure/fastly-vcl-badreferer.html
functional_areas:
  - Cloud
  - Setup
---

> In some cases, you might need to bypass Fastly caching and submit requests directly to the origin server, for example to determine whether site issues are caused by caching, or to troubleshoot  You can use the VCL configure a custom VCL snippet to bypass Fastly caching and submit requests directly to the origin server. This might be to 

Author: ```suggestion
You can create a custom VCL snippet to bypass Fastly caching if you need to submit requests directly to the origin server, for example to determine whether site issues are caused by caching, or to troubleshoot headers. You can configure the snippet to bypass Fastly for requests from a specific IP address or URL.
```

{:.bs-callout-info}
We recommend testing custom VCL configurations in a Staging environment before merging them into a Production environment.

{:.procedure}
Prerequisites

{%include cloud/cloud-fastly-prereqs-custom-vcl.md%}

{:.procedure}

> To bypass Fastly and submit requests to the Origin server:

Author: ```suggestion
To bypass Fastly and submit requests to the origin server:
```

{% include cloud/admin-ui-login-step.md %}

1. Click **Stores** > Settings > **Configuration** > **Advanced** > **System**.

1. Expand **Full Page Cache** > **Fastly Configuration** > **Custom VCL Snippets**.

1. Click **Create Custom Snippet**.

1. Add the VCL snippet values:

   -  **Name** — `bypass_fastly`

   -  **Type** — `recv`

   -  **Priority** — `5`

   -  **VCL** snippet content —

      The following example bypasses Fastly for a specific IP address:

      ```conf
      if (client.ip == "<Your IPv4 IP address>" || client.ip == "<Your IPv6 IP address>") {
        return(pass);
      }
      ```

      The following example bypasses Fastly for a specific URL pattern:

      ```conf

>       if (req.url ~ "/media/feeds/GoogleShoppingHiVisNew.xml") {  return (pass);}

Author: ```suggestion
      if (req.url ~ "/media/feeds/GoogleShoppingHiVisNew.xml")  {
         return (pass);
      }
```
